.NET 6 Preview 7 Blazor WebAssembly won't build if you are using a JsonSerializerContext implementation

In preview 6, you introduced the Json source generator, and I created a "MyJsonContext" to enable the source generator.

```c#
    [JsonSerializable(typeof(JsonData))]
    internal partial class MyJsonContext : JsonSerializerContext
    {
    }
```

Now, in preview 7 the System.Text.Json.SourceGeneration assembly gets added to references twice.

![image](https://user-images.githubusercontent.com/11338430/129119631-3fae9148-027d-46b8-a378-03daa95d4e1b.png)

This leads to build failure as they both generate the same code!

You can see the duplicate reference by simply creating a new BlazorWebAssembly project, but you need to implement a JsonSerializerContext to see the errors.

### Configuration


.NET version 6.0.100-preview.7.21379.14
Windows 11 22000.120
Architecture x64
Browser not relevant as the project won't build.

### Regression?

Worked in .NET 6 Preview 6

### Other information

Errors from build:

1>Build FAILED.
1>
1>C:\Users\aaaaa\source\repos\JsonPerfTest\System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\MyJsonContext.g.cs(6,6,6,60): error CS0579: Duplicate 'global::System.CodeDom.Compiler.GeneratedCodeAttribute' attribute
1>C:\Users\aaaaa\source\repos\JsonPerfTest\System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\MyJsonContext.String.g.cs(7,101,7,108): error CS0102: The type 'MyJsonContext' already contains a definition for '_String'
1>C:\Users\aaaaa\source\repos\JsonPerfTest\System.Text.Json.SourceGeneration\System.Text.Json.SourceGeneration.JsonSourceGenerator\MyJsonContext.String.g.cs(8,100,8,106): error CS0102: The type 'MyJsonContext' already contains a definition for 'String'
...68 of these errors in total

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets
@@ -369,6 +369,12 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     </ResolveTargetingPackAssets>
 
+    <PropertyGroup>
+      <!-- Unescape this semicolon-delimited ordered list so that it's passed as a list to targets
+           that consume it. -->
+      <PackageConflictPreferredPackages>$([MSBuild]::Unescape($(PackageConflictPreferredPackages)))</PackageConflictPreferredPackages>
+    </PropertyGroup>
+
     <ItemGroup Condition="'$(RuntimeIdentifier)' == '' or '$(SelfContained)' != 'true'">
       <PackageConflictPlatformManifests Include="@(PlatformManifestsFromTargetingPacks)" />
     </ItemGroup>

--- a/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs
+++ b/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs
@@ -1061,5 +1061,29 @@ namespace ProjectNameWithSpaces
                 .And
                 .HaveStdOutContaining("NETSDK1148");
         }
+
+        [Fact]
+        public void It_Has_Unescaped_PackageConflictPreferredPackages_Values()
+        {
+            string targetFramework = ToolsetInfo.CurrentTargetFramework;
+
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("AppWithLibrary", identifier: targetFramework)
+                .WithSource()
+                .WithTargetFramework(targetFramework, "TestLibrary");
+
+            var getValuesCommand = new GetValuesCommand(Log, Path.Combine(testAsset.TestRoot, "TestLibrary"), targetFramework, "PackageConflictPreferredPackages");
+            getValuesCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            List<string> preferredPackages = getValuesCommand.GetValues();
+            preferredPackages.Should().NotBeEmpty();
+            preferredPackages.Count.Should().BeGreaterThan(1);
+
+            preferredPackages.Should().NotContain(packageName => packageName.Contains(';'),
+                because: "No package name should have a semicolon in it--PackageConflictPreferredPackages should be a semicolon delimited list of package names");
+        }
     }
 }

